### PR TITLE
trozler refactor3

### PR DESCRIPTION
**Pull request by trozler**

### CODE REVIEW

File: contracts/WhiteElephantFactory.sol [hunk 1]
```diff
@@ -2,17 +2,26 @@
...
+
   TODO: 
-  1. Check create game flow.
-  2. Check join game flow.
   3. Check deposit token flow. 
-  4. Check refund gift button.
-  5. Check start game flow.
 */
-contract WhiteElephantFactory {
+contract WhiteElephantFactory is Ownable {
```
Comment (lorenzoPrimi) [inline contracts/WhiteElephantFactory.sol:22]: Was this removed or never here? good catch!

File: contracts/WhiteElephantFactory.sol [hunk 2]
```diff
@@ -21,75 +30,72 @@ contract WhiteElephantFactory {
...
     }
 
     constructor() {
         emit FactoryCreated(msg.sender, block.timestamp);
     }
 
-    function createGame(
+    /**
+     * @dev creating game and depositing a token should be atomic.
+     * When you create game you also need to deposit.
```
Comment (lorenzoPrimi) [inline contracts/WhiteElephantFactory.sol:58]: The creator shouldn't be forced to be a participant of the game. To review this one

File: contracts/WhiteElephantGame.sol [hunk 3]
```diff
@@ -157,133 +151,151 @@ contract WhiteElephantGame is Ownable {
...
         (
             uint80 roundID,
             int256 price,
             uint256 startedAt,
             uint256 timeStamp,
             uint80 answeredInRound
         ) = priceFeed.latestRoundData();
+     */
+    //TODO pass as argument the token name / address
+    function _getPrice() private view returns (int256) {
```
Comment (lorenzoPrimi) [inline contracts/WhiteElephantGame.sol:282]: Remove this function or merge the chainlink branch

File: contracts/WhiteElephantFactory.sol (hunk 2 above)
Comment (trozler) [reply contracts/WhiteElephantFactory.sol:58]: I agree will change.

File: contracts/WhiteElephantFactory.sol (hunk 1 above)
Comment (trozler) [reply contracts/WhiteElephantFactory.sol:22]: I'm not sure but should be there.

**Pull request merged**